src/utils/manager.py

```diff
@@ -59,18 +53,41 @@ class MountInfo:
     passno: str
 
 
-def _translate(url: str) -> Tuple[str, str]:
+def _translate(url: str) -> Optional[Tuple[str, Optional[int]]]:
     """Translate an NFS URL to a mount.nfs endpoint.
 
     Args:
         url: NFS uniform resource locator (URL)
 
     Returns:
-        Tuple[str, str]: `mount.nfs`-understandable endpoint and port number.
+        Optional[Tuple[str, Optional[int]]: `mount.nfs`-understandable endpoint and port number.
+        `None` if the url was invalid.
     """
-    _logger.debug(f"Converting NFS URL {url} to `mount.nfs` format")
-    nfs_url = urlparse(url)
-    return f"{nfs_url.hostname}:{nfs_url.path}", nfs_url.port
+    _logger.debug(f"Translating NFS URL {url} to `mount.nfs` format")
+    try:
+        nfs_url = urlsplit(url)
+    except ValueError:
+        return None
+
+    if not (host := nfs_url.hostname):
+        return None
+
+    try:
+        port = nfs_url.port
+    except ValueError:
+        return None
+
+    if nfs_url.fragment or nfs_url.query or nfs_url.scheme != "nfs":
+        return None
+
+    # Only IPv6 addresses require square brackets.
+    try:
+        IPv6Address(host)
+        return f"[{host}]:{nfs_url.path}", port
+    except AddressValueError:
+        pass
+
+    return f"{host}:{nfs_url.path}", port
```

Comment:
Since the `.port` descriptor can raise a `ValueError` if the port is invalid in the NFS URL, why don't we catch two birds in one net :goal_net:?

Given that our desired behavior here is to just return `None` if a `ValueError` is raised here, we can just handle both within the same try-except block. We could reduce this block down to the following:

```python3
try:
    nfs_url = urlsplit(url)
    host = nfs_url.hostname
    port = nfs_url.port
except ValueError:
    return

if not host or nfs_url.fragment or nfs_url.query or nfs_url.scheme != "nfs":
    return

try:
    # IPv6 address require square brackets.
    IPv6Address(host)
    return f"[{host}]:{nfs_url.path}", port
except AddressValueError:
    # Hostname is not an IPv6 address.
    return f"{host}:{nfs_url.path}", path
```

A few less LOC for us to worry about. Also, just a couple side notes:

* If no return value is given to `return`, `return` will return `None` by default. We don't need to worry about carrying `return None`.
* Since an `AddressValueError` means that we'll just go ahead and return the mount path without square brackets, rather than just passing out of the try-except block, let's just return the mount endpoint and path.

Let me know if you're good with this.

Reply:
Yep, looks cleaner overall.